packages/components/src/Button/Button.tsx

```diff
@@ -51,11 +66,24 @@ const _Button = forwardRef<HTMLButtonElement, ButtonProps>(
         className={cn(
           'inline-flex items-center justify-center gap-[0.5ch]',
           classNames,
-          fullWidth ? 'w-full' : undefined
+          fullWidth ? 'w-full' : undefined,
+          loading && 'pointer-events-none !cursor-progress'
         )}
+        isPending={loading}
         isDisabled={disabled}
       >
-        {children}
+        {loading ? (
+          <>
+            <span className="absolute">
+              <ProgressCycle />
+            </span>
+            <span className="invisible flex items-center gap-2 opacity-0">
```

Comment:
either `invisible` or `opactiy`?

Also: hardcoding the gap etc here might get us in trouble if these things change in the theme (e.g. default is `gap-[0.5ch]`).

This might work (also centers the loading spinner :P):
https://play.tailwindcss.com/09rhHtzot3